examples/remarketing/upload_conversion_adjustment.py

```diff
@@ -0,0 +1,125 @@
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This example imports conversion adjustments for existing conversions.
+
+To set up a conversion action, run the add_conversion_action.py example.
+"""
+
+
+import argparse
+import sys
+
+from google.ads.google_ads.client import GoogleAdsClient
+from google.ads.google_ads.errors import GoogleAdsException
+
+
+def main(client, customer_id, conversion_action_id, gcl_id, adjustment_type,
+         conversion_time, adjustment_time, restatement_value):
+    # Determine the adjustment type.
+    conversion_adjustment_type_enum = (
+        client.get_type('ConversionAdjustmentTypeEnum')
+    )
+    if adjustment_type.lower() == 'retraction':
+        conversion_adjustment_type = conversion_adjustment_type_enum.RETRACTION
+    elif adjustment_type.lower() == 'restatement':
+        conversion_adjustment_type = conversion_adjustment_type_enum.RESTATEMENT
+    else:
+        raise ValueError('Invalid adjustment type specified.')
+
+    # Create the conversion adjustment and set basic values.
+    conversion_adjustment = (client.get_type('ConversionAdjustment',
+                                             version='v2'))
+    conversion_action_service = (client.get_service('ConversionActionService',
+                                                   version='v2'))
+    conversion_adjustment.conversion_action.value = (
+        conversion_action_service.conversion_action_path(
+            customer_id, conversion_action_id)
+    )
+    conversion_adjustment.adjustment_type = conversion_adjustment_type
+    conversion_adjustment.adjustment_date_time.value = adjustment_time
+
+    # If the restatement value is specified, set it (default currency is USD).
+    if (restatement_value and
+        conversion_adjustment_type ==
+        conversion_adjustment_type_enum.RESTATEMENT):
+        conversion_adjustment.restatement_value.adjusted_value.value = (
+            float(restatement_value))
+        conversion_adjustment.restatement_value.currency_code.value = 'USD'
+
+    # Set the Gclid Date
+    conversion_adjustment.gclid_date_time_pair.gclid.value = gcl_id
+    conversion_adjustment.gclid_date_time_pair.conversion_date_time.value = (
+        conversion_time)
+
+    # Try the upload
+    conversion_adjustment_upload_service = (
+        client.get_service('ConversionAdjustmentUploadService', version='v2')
+    )
+
+    try:
+        response = (
+            conversion_adjustment_upload_service.
+            upload_conversion_adjustments(customer_id,
+                                           [conversion_adjustment],
+                                           partial_failure=True)
```

Comment:
Indentation of arguments on following lines are off-by-1.

Reply:
fixed